### PR TITLE
Test iOS cross-compile in CI

### PR DESCRIPTION
**Pull request by kelson42**
Test macOS to iOS cross-compilation in CI

### CODE REVIEW

**Comment by kelson42** [comment]
@mgautierfr @rgaudin `meson_cross_file.txt` seems to be missing. An other point is that if we really need `-Db_coverage=true` here?!

**Comment by rgaudin** [comment]
> @mgautierfr @rgaudin `meson_cross_file.txt` seems to be missing. An other point is that if we really need `-Db_coverage=true` here?!

It's downloading https://tmp.kiwix.org/ci/deps2_macos_native_dyn_libkiwix.tar.xz which obviously (so I think) doesn't contain `BUILD_iOS_xxxx/` folder (which would contain the cross file.

**Comment by kelson42** [comment]
@rgaudin Thx, indeed!

Now it seems the `libzim` is not part of the deps! See https://github.com/kiwix/libkiwix/actions/runs/6835979771/job/18590377587?pr=1018#step:5:71

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,21 +33,30 @@ jobs:
...
           brew install pkg-config ninja meson
 
-      - name: Install dependencies
-        env:
-          ARCHIVE_NAME: deps2_macos_native_dyn_libkiwix.tar.xz
+      - name: Install deps
+        shell: bash
         run: |
-          wget -O- https://tmp.kiwix.org/ci/${{env.ARCHIVE_NAME}} | tar -xJ -C ${{env.HOME}}
+          ARCHIVE_NAME=deps2_macos_${{matrix.target}}_libzim.tar.xz
```
Comment (mgautierfr) [inline .github/workflows/ci.yml:39]: `deps2_*_libzim.tar.xz` is dependencies for `libzim`. Probably a wrong copy/paste :)

**Comment by rgaudin** [comment]
Changed `libzim` suffix to `libkiwix` but in fact we don't produce those deps for the other archs: macOS arm64 (not listed), iOS x86_64 and iOS arm64

The only macos `deps2` we produce for libkiwix is [`deps2_macos_native_dyn_libkiwix.tar.xz`](https://tmp.kiwix.org/ci/deps2_macos_native_dyn_libkiwix.tar.xz)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kelson42) [reply .github/workflows/ci.yml]: OK, sorry for this stupid thing!

**Comment by kelson42** [comment]
@mgautierfr Can we easi

> The only macos `deps2` we produce for libkiwix is [`deps2_macos_native_dyn_libkiwix.tar.xz`](https://tmp.kiwix.org/ci/deps2_macos_native_dyn_libkiwix.tar.xz)

@mgautierfr Can we easily fix this?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,21 +33,30 @@ jobs:
...
 
-      - name: Compile source code
+      - name: Compile
         env:
           PKG_CONFIG_PATH: ${{env.HOME}}/BUILD_native_dyn/INSTALL/lib/pkgconfig
           CPPFLAGS: -I${{env.HOME}}/BUILD_native_dyn/INSTALL/include
+        shell: bash
         run: |
-          meson . build --default-library=shared -Db_coverage=true
+          MESON_OPTION="--default-library=shared -Db_coverage=true"
```
Comment (rgaudin) [inline .github/workflows/ci.yml:48]: I find it more maintainable to not use github contexts in `run`. I prefer to pass all values as env instead:

- easier to test your code locally. You just have to prepare your exports and then copy-paste.
- Github logs prints the env always so it's easy to spot any mistake or missing value
- It looks like *dependencies* for the command which is good for renames/cleanups

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -23,21 +33,30 @@ jobs:
...
+          MESON_CROSSFILE="$HOME/BUILD_${{matrix.target}}/meson_cross_file.txt"
+          if [[ ! "${{matrix.target}}" =~ native_.* ]]; then
+            MESON_OPTION="$MESON_OPTION --cross-file $MESON_CROSSFILE -Dstatic-linkage=true"
+            cat $MESON_CROSSFILE
```
Comment (rgaudin) [inline .github/workflows/ci.yml:52]: Do you want to keep it? I think it's useful but it looks like a debug line 😀

**Pull request merged**